### PR TITLE
chore(wdio): migrate slot-conditional-rendering to wdio

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

port the `test/karma/test-app/slot-conditional-rendering` test suite to webdriverio

git history has been preserved as much as possible here, using `git mv` to move files from the `karma` dir to `wdio/` in all possible cases. however, git may fail to mark some files as 'moved' if we change a file enough that it exceeds git's "sameness threshold".

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

I made each of the tests fail as I converted them.

## Other information
STENCIL-1191

### CODE REVIEW

File: test/wdio/slot-conditional-rendering/cmp.test.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    await expect($('#slotted-content-element-id')).not.toHaveAttribute('hidden');
+  });
+
+  it('header slot becomes hidden after hit the toggle button', async () => {
+    await expect($('#slotted-header-element-id')).not.toHaveAttribute('hidden');
+
+    await $('#header-visibility-toggle').click();
+    await browser.waitUntil(() => {
+      return document.querySelector('#slotted-header-element-id').hasAttribute('hidden');
+    });
```
Comment (christian-bromann) [inline test/wdio/slot-conditional-rendering/cmp.test.tsx:31]: This should not be needed as the `toHaveAttribute` matcher will retry the condition until it's met.
```suggestion
```

File: test/wdio/slot-conditional-rendering/cmp.test.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    await expect($('#slotted-header-element-id')).toHaveAttribute('hidden');
+  });
+
+  it('content slot becomes hidden after hit the toggle button', async () => {
+    await expect($('#slotted-content-element-id')).not.toHaveAttribute('hidden');
+
+    await $('#content-visibility-toggle').click();
+    await browser.waitUntil(() => {
+      return document.querySelector('#slotted-content-element-id').hasAttribute('hidden');
+    });
```
Comment (christian-bromann) [inline test/wdio/slot-conditional-rendering/cmp.test.tsx:42]: same
```suggestion
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]

**Review by alicewriteswrongs — APPROVED** [review]
looks good other than @christian-bromann's comments

File: test/wdio/slot-conditional-rendering/cmp.test.tsx (hunk 1 above)
Comment (rwaskiewicz) [reply test/wdio/slot-conditional-rendering/cmp.test.tsx]: Nice! Fixed in 82a878c1d

**Review by christian-bromann — APPROVED** [review]

**Pull request merged**